File: binary_search/problems/src/findMinRot.go
package main

import (
	"fmt"
)

func main() {
	nums := []int{2, 2, 2, 0, 1}
	min := findMin(nums)
	fmt.Println("Minimum element in: ", nums, " is: ", min)
}

func findMin(nums []int) int {
	left, right := 0, len(nums)-1
	for left < right {
		mid := (left + right) / 2
		if nums[mid] > nums[right] {
			left = mid + 1
		} else if nums[mid] < nums[right] {
			right = mid
		} else {
			right--
		}
	}
	return nums[left]
}

File: binary_search/problems/findMinRot.py
def find_min(nums):
  left, right = 0, len(nums) - 1
  while left < right:
    mid = (left + right) // 2
    if nums[mid] > nums[right]:
      left = mid + 1
    elif nums[mid] < nums[right]:
      right = mid
    else:
      right -= 1
  return nums[left]

print(find_min([2,2,2,0,1]))